### ISSUE TITLE
Error on loading 

### ISSUE THREAD

**Issue by benwestwood** [MiniKeePass/MiniKeePass#567]
The app isn't opening. It's giving me an error after I type in my password. "Error Could not open database

**Comment by wireddude** [comment]
Suddenly seeing this as well. Would adding a new account with an unsupported attachment type cause this to happen ? The same file opens just fine on all my other clients.

**Comment by wireddude** [comment]
Also, using a keyfile. Not sure of that matters

**Comment by chaosbunker** [comment]
Same here. Yesterday it still worked for me. I was using PW + Keyfile. Around 12 hours ago I couldn't open the database anymore. It's confusing, because I don't think I changed anything ...

Even a database without keyfile won't open anymore.

**Comment by wireddude** [comment]
Pull request #568 related to this?

**Comment by chaosbunker** [comment]
No, not related. Works fine again. It is possible, that the database could temporarily not be opened on my iPhone because the .lock-file was also synced to the phone. I made sure this does not happen anymore and the problem seems resolved for now.

I also switched from KeePassXC on mac to [MacPass](https://github.com/mstarke/MacPass), which does not create a lock file.

**Comment by chaosbunker** [comment]
So this happened again and I think it might be related to #352 

I saw in the console of XCode that it says "InvalidFileSignature". The database opens just fine on my mac with the same password and keyfile. I exported the database from MacPass and saved as a new kdbx file and it works again on the phone

**Comment by chaosbunker** [comment]
Another update: The same database that did not work anyomore suddenly works again, after adding more items to it. I accidentally kept using the database that could not be opened on the phone and noticed that after syncing the updated version, it works again. I guess when what I reported in [this previous comment](https://github.com/MiniKeePass/MiniKeePass/issues/567#issuecomment-338490671) had nothing to do with the .lock file, but was also because of the same issue.

How can this be investigated? I really like miniKeePass, but the possibility that I suddenly don't have access to the database anymore on the phone is a bit of a problem :-/

**Comment by numo68** [comment]
- imported a database from KeePass 2.38 on Windows, works
- edited the same database in KeepassXC 2.3.0, imported, Could not open database
- edited the same database in KeePass 2 again, imported, works

No keyfile, just password. The database edited by KeePassXC can be opened in KeePass 2 just fine, just the MiniKeePass is not able to open it.

**Comment by KnuckleBob** [comment]
I've run into this issue as well (Error: Could not open database), running KeePassXC on Mac.

**Comment by ghost** [comment]
For me this happened when I tried any Key Derivation Function other than `AES-KDF (KDBX 3.1)`

**Comment by ghost** [comment]
For anyone running into this issue, the Encryption Algo must be `AES: 256-bit` and the Key Derivation Function must be `AES-KDF (KDBX 3.1)`

**Comment by numo68** [comment]
My database uses AES 256 and AES-KDF, yet I am experiencing the issue. As I really doubt that editing the db in KeepassXC changes these parameters the culprit has to be something different.

**Comment by ghost** [comment]
Did you set it to `AES-KDF (KDBX 3.1)`? I can confirm that the KDBX 4.0 version definitely doesn't work.

I also had to disable compression.

**Comment by numo68** [comment]
I don't have the KeepassXC installed right now so I cannot verify what exactly it is showing. The Keepass 2 does not show version numbers, it just allows to choose between AES-KDF and Argon2. The Argon2 was indeed added in the KDBX 4 format version.

But again, it was the _same_ database file, just opened and edited in different applications. This operation _cannot_ change the KDF used, the shuffling of the passphrase has to arrive at the same key. The KeepassXC might write something differently to the header but I am pretty sure the algorithm used is the one used in the kdbx 3.1.

If the KeepassXC changes the kdbx format to 4.0 without telling anyone, would the Keepass 2  change it _back_ to 3.1? Possible, but not very probable, IMHO. A new editing in Keepass 2 however makes the file readable again.

I have the GZip compression enabled in KeePass 2 and this was never a problem. It might of course be that the KeePassXC has a bug there.

Anyway, the last commit was 6 months ago, as was the last version in the app store, and the developers don't seem to be interested in this issue so I'm not holding my breath...

**Comment by bendauphinee** [comment]
Confirmed swapping to AES 256 and AES-KDF is working. (Was using AES 256 and Argon2)

**Comment by upsatish** [comment]
Hi
I believe a database upgrade to KDBX 4 may have happened when KeePassXC was upgraded. KeePassXC 2.2.4 cannot open the upgraded database. KeePassXC 2.3.0 can. Obviously MiniKeePass does not support this.

Fix:
I used KeePass 2.42.1 (original) on Windows to downgrade my database:
Click on File -> Export -> KeePass KDBX (2.34, Old Format)
and save to a new file to export to the older KDBX format. Hope this helps.

It now works on KeePassXC 2.2.4 and MiniKeePass. I will use KeePassXC 2.2.4 till MiniKeePass is updated to accomodate KDBX 4, KeePass 2.35 and above formats. Alternative is to use Strongbox on iOS.

Changing the database encryption algorithm did not fix my issue.

KeePass 2.42.1 runs on Mono if you have Mac or Linux.

See also:
https://github.com/MiniKeePass/MiniKeePass/issues/612
https://github.com/MiniKeePass/MiniKeePass/issues/577
